GetResultTests.testToAndFromXContentEmbedded failure

**Build scan**: N/A, local build. 

**Repro line**:
./gradlew ':server:test' --tests "org.elasticsearch.index.get.GetResultTests.testToAndFromXContentEmbedded" -Dtests.seed=E5FC519FC1275FEB -Dtests.security.manager=true -Dtests.locale=ar-AE -Dtests.timezone=Africa/Blantyre -Druntime.java=15

**Reproduces locally?**:
Yes

**Applicable branches**:
master

**Failure excerpt**:

<details><summary>Failure Details</summary>
<p>

```
Suite: Test class org.elasticsearch.index.get.GetResultTests
  1> [2021-01-06T03:49:14,797][INFO ][o.e.i.g.GetResultTests   ] [testToAndFromXContentEmbedded] before test
  1> [2021-01-06T03:49:15,006][INFO ][o.e.i.g.GetResultTests   ] [testToAndFromXContentEmbedded] after test
  2> REPRODUCE WITH: ./gradlew ':server:test' --tests "org.elasticsearch.index.get.GetResultTests.testToAndFromXContentEmbedded" -Dtests.seed=E5FC519FC1275FEB -Dtests.security.manager=true -Dtests.locale=ar-AE -Dtests.timezone=Africa/Blantyre -Druntime.java=15
  2> java.lang.AssertionError: expected:<{
      "_index" : null,
      "_id" : null,
      "_seq_no" : 193165536962485811,
      "_primary_term" : 71,
      "_nested_path" : "Rmhd",
      "_routing" : "VYZsPS",
      "found" : true,
      "_source" : {
        "cDdkOed" : {
          "kIOmfBuye" : -4355782572476556875
        },
        "CHPTRkb" : 0.1974391328248285,
        "ngifuk" : "SIisj",
        "YjVWYCSNxE" : {
          "fErBpnn" : [
            {
              "pgdJSUie" : "KYm",
              "LAMHSFzle" : 0.5538381620236812
            },
            {
              "xJWtzXYOzq" : 2501515057617439655,
              "uILBlbRl" : "CYOBNHnQgQ",
              "kPYyInFAX" : 0.7238725410136407,
              "kegvNLXO" : -6906501997487968558,
              "QRRCPof" : 0.06681025923342088
            },
            {
              "EWvLpm" : "ZYBSyKZl",
              "VKbUdec" : 0.8878569548237388,
              "mkvlDUag" : 0.035264934026060124,
              "cysyqBfZvG" : "fCnF",
              "DWiIKgq" : "duJkfxQLUh"
            },
            {
              "XpiKBYLTYV" : 0.8365711755598817,
              "nEDSFe" : "NwGW",
              "gPlQFoaTqV" : 0.6106422641750852,
              "KApbrEZv" : -3187916961566692938,
              "YsLfTgF" : "HptyaRpTW"
            },
            {
              "EtCSBHjKHV" : 2032070882424922864,
              "XjtZKWu" : "aluUN",
              "cdsLri" : 646340918915008831,
              "xrYZLhjvg" : "qupmRsWQoA"
            }
          ]
        }
      },
      "fields" : {
        "dFszYoJZbp" : [
          "7qCb6oSJ8bGYgtic1pRLFNWN5Yu+8baHptebCvGpmaXzk5it6ZqjUee7rfKViLRdBMyJ3qfzg42AyZfun4nziLm17qK+8724hs+t7o2E7oSv7riW6KWnH+Woqe6ihe+MmOSFo3A="
        ],
        "jTooydPG" : [
          0.4488774315814221,
          0.23740623389882665,
          0.6111549990330503,
          0.6939449725143741,
          0.4813573856150549
        ],
        "POYUYSDipG" : [
          "lIT"
        ],
        "FQirH" : [
          {
            "pzrOO" : false,
            "xpySH" : "PrsvrkaFxc",
            "Nfuue" : -445787976
          }
        ],
        "ajOPP" : [
          {
            "PxdIx" : 515179785,
            "jYEEp" : "aAmcBIYApR",
            "BqGTY" : false
          },
          {
            "JRerP" : "xjecjpuTHz",
            "ONjWt" : 833395824,
            "tcZYJ" : false
          },
          {
            "Rfyki" : 886453855,
            "fcNqw" : false,
            "obrYj" : "FDxlDkRXFo"
          },
          {
            "fzSZH" : 1344392580,
            "vsHAr" : false,
            "NkxTf" : "FKZZYrNqIn"
          },
          {
            "dSMsD" : "zyTpYzfuqT",
            "wzqdI" : 1140600156,
            "CXNJw" : true
          }
        ]
      }
    }> but was:<{
      "_index" : null,
      "_id" : null,
      "_seq_no" : 193165536962485811,
      "_primary_term" : 71,
      "_routing" : "VYZsPS",
      "_nested_path" : "Rmhd",
      "found" : true,
      "_source" : {
        "cDdkOed" : {
          "kIOmfBuye" : -4355782572476556875
        },
        "CHPTRkb" : 0.1974391328248285,
        "ngifuk" : "SIisj",
        "YjVWYCSNxE" : {
          "fErBpnn" : [
            {
              "pgdJSUie" : "KYm",
              "LAMHSFzle" : 0.5538381620236812
            },
            {
              "xJWtzXYOzq" : 2501515057617439655,
              "uILBlbRl" : "CYOBNHnQgQ",
              "kPYyInFAX" : 0.7238725410136407,
              "kegvNLXO" : -6906501997487968558,
              "QRRCPof" : 0.06681025923342088
            },
            {
              "EWvLpm" : "ZYBSyKZl",
              "VKbUdec" : 0.8878569548237388,
              "mkvlDUag" : 0.035264934026060124,
              "cysyqBfZvG" : "fCnF",
              "DWiIKgq" : "duJkfxQLUh"
            },
            {
              "XpiKBYLTYV" : 0.8365711755598817,
              "nEDSFe" : "NwGW",
              "gPlQFoaTqV" : 0.6106422641750852,
              "KApbrEZv" : -3187916961566692938,
              "YsLfTgF" : "HptyaRpTW"
            },
            {
              "EtCSBHjKHV" : 2032070882424922864,
              "XjtZKWu" : "aluUN",
              "cdsLri" : 646340918915008831,
              "xrYZLhjvg" : "qupmRsWQoA"
            }
          ]
        }
      },
      "fields" : {
        "FQirH" : [
          {
            "pzrOO" : false,
            "xpySH" : "PrsvrkaFxc",
            "Nfuue" : -445787976
          }
        ],
        "dFszYoJZbp" : [
          "7qCb6oSJ8bGYgtic1pRLFNWN5Yu+8baHptebCvGpmaXzk5it6ZqjUee7rfKViLRdBMyJ3qfzg42AyZfun4nziLm17qK+8724hs+t7o2E7oSv7riW6KWnH+Woqe6ihe+MmOSFo3A="
        ],
        "jTooydPG" : [
          0.4488774315814221,
          0.23740623389882665,
          0.6111549990330503,
          0.6939449725143741,
          0.4813573856150549
        ],
        "POYUYSDipG" : [
          "lIT"
        ],
        "ajOPP" : [
          {
            "PxdIx" : 515179785,
            "jYEEp" : "aAmcBIYApR",
            "BqGTY" : false
          },
          {
            "JRerP" : "xjecjpuTHz",
            "ONjWt" : 833395824,
            "tcZYJ" : false
          },
          {
            "Rfyki" : 886453855,
            "fcNqw" : false,
            "obrYj" : "FDxlDkRXFo"
          },
          {
            "fzSZH" : 1344392580,
            "vsHAr" : false,
            "NkxTf" : "FKZZYrNqIn"
          },
          {
            "dSMsD" : "zyTpYzfuqT",
            "wzqdI" : 1140600156,
            "CXNJw" : true
          }
        ]
      }
    }>
        at __randomizedtesting.SeedInfo.seed([E5FC519FC1275FEB:B5959AD1926D8B93]:0)
        at org.junit.Assert.fail(Assert.java:88)
        at org.junit.Assert.failNotEquals(Assert.java:834)
        at org.junit.Assert.assertEquals(Assert.java:118)
        at org.junit.Assert.assertEquals(Assert.java:144)
        at org.elasticsearch.index.get.GetResultTests.testToAndFromXContentEmbedded(GetResultTests.java:120)
```
</p>
</details>

--- a/client/rest-high-level/src/main/java/org/elasticsearch/client/enrich/PutPolicyRequest.java
+++ b/client/rest-high-level/src/main/java/org/elasticsearch/client/enrich/PutPolicyRequest.java
@@ -80,7 +80,7 @@ public final class PutPolicyRequest implements Validatable, ToXContentObject {
 
     // package private for testing only
     void setQuery(BytesReference query) {
-        assert query == null || XContentHelper.xContentType(query) == XContentType.JSON :
+        assert query == null || XContentHelper.xContentType(query).canonical() == XContentType.JSON :
                 "Only accepts JSON encoded query but received [" + Strings.toString(query) + "]";
         this.query = query;
     }

--- a/server/src/main/java/org/elasticsearch/common/xcontent/XContentHelper.java
+++ b/server/src/main/java/org/elasticsearch/common/xcontent/XContentHelper.java
@@ -209,7 +209,7 @@ public class XContentHelper {
     public static String convertToJson(BytesReference bytes, boolean reformatJson, boolean prettyPrint, XContentType xContentType)
         throws IOException {
         Objects.requireNonNull(xContentType);
-        if (xContentType == XContentType.JSON && !reformatJson) {
+        if (xContentType.canonical() == XContentType.JSON && !reformatJson) {
             return bytes.utf8ToString();
         }
 

--- a/server/src/test/java/org/elasticsearch/common/xcontent/XContentFactoryTests.java
+++ b/server/src/test/java/org/elasticsearch/common/xcontent/XContentFactoryTests.java
@@ -49,6 +49,13 @@ public class XContentFactoryTests extends ESTestCase {
         testGuessType(XContentType.CBOR);
     }
 
+    public void testGuessVndTypes() throws IOException {
+        testGuessType(XContentType.VND_JSON);
+        testGuessType(XContentType.VND_SMILE);
+        testGuessType(XContentType.VND_YAML);
+        testGuessType(XContentType.VND_CBOR);
+    }
+
     private void testGuessType(XContentType type) throws IOException {
         XContentBuilder builder = XContentFactory.contentBuilder(type);
         builder.startObject();
@@ -56,7 +63,7 @@ public class XContentFactoryTests extends ESTestCase {
         builder.endObject();
 
         final BytesReference bytes;
-        if (type == XContentType.JSON && randomBoolean()) {
+        if (type.canonical() == XContentType.JSON && randomBoolean()) {
             final int length = randomIntBetween(0, 8 * XContentFactory.GUESS_HEADER_LENGTH);
             final String content = Strings.toString(builder);
             final StringBuilder sb = new StringBuilder(length + content.length());
@@ -68,12 +75,12 @@ public class XContentFactoryTests extends ESTestCase {
             bytes = BytesReference.bytes(builder);
         }
 
-        assertThat(XContentHelper.xContentType(bytes), equalTo(type));
-        assertThat(XContentFactory.xContentType(bytes.streamInput()), equalTo(type));
+        assertThat(XContentHelper.xContentType(bytes), equalTo(type.canonical()));
+        assertThat(XContentFactory.xContentType(bytes.streamInput()), equalTo(type.canonical()));
 
         // CBOR is binary, cannot use String
-        if (type != XContentType.CBOR && type != XContentType.SMILE) {
-            assertThat(XContentFactory.xContentType(Strings.toString(builder)), equalTo(type));
+        if (type.canonical() != XContentType.CBOR && type.canonical() != XContentType.SMILE) {
+            assertThat(XContentFactory.xContentType(Strings.toString(builder)), equalTo(type.canonical()));
         }
     }
 

--- a/test/framework/src/main/java/org/elasticsearch/test/RandomObjects.java
+++ b/test/framework/src/main/java/org/elasticsearch/test/RandomObjects.java
@@ -136,7 +136,7 @@ public final class RandomObjects {
      */
     public static Object getExpectedParsedValue(XContentType xContentType, Object value) {
         if (value instanceof BytesArray) {
-            if (xContentType == XContentType.JSON) {
+            if (xContentType.canonical() == XContentType.JSON) {
                 //JSON writes base64 format
                 return Base64.getEncoder().encodeToString(((BytesArray)value).toBytesRef().bytes);
             }

--- a/test/framework/src/main/java/org/elasticsearch/test/rest/yaml/ClientYamlTestResponse.java
+++ b/test/framework/src/main/java/org/elasticsearch/test/rest/yaml/ClientYamlTestResponse.java
@@ -126,7 +126,8 @@ public class ClientYamlTestResponse {
     public String getBodyAsString() {
         if (bodyAsString == null && body != null) {
             //content-type null means that text was returned
-            if (bodyContentType == null || bodyContentType == XContentType.JSON || bodyContentType == XContentType.YAML) {
+            if (bodyContentType == null || bodyContentType.canonical() == XContentType.JSON ||
+                bodyContentType.canonical() == XContentType.YAML) {
                 bodyAsString = new String(body, StandardCharsets.UTF_8);
             } else {
                 //if the body is in a binary format and gets requested as a string (e.g. to log a test failure), we convert it to json

--- a/x-pack/plugin/watcher/src/test/java/org/elasticsearch/xpack/watcher/support/WatcherTemplateTests.java
+++ b/x-pack/plugin/watcher/src/test/java/org/elasticsearch/xpack/watcher/support/WatcherTemplateTests.java
@@ -45,7 +45,7 @@ public class WatcherTemplateTests extends ESTestCase {
     }
 
     public void testEscaping() throws Exception {
-        XContentType contentType = randomFrom(XContentType.values());
+        XContentType contentType = randomFrom(XContentType.values()).canonical();
         if (rarely()) {
             contentType = null;
         }

